### PR TITLE
fix mkdtemp

### LINKED ISSUE #1144: Segmentation fault using await mkdtemp
### Version

0.1.10

### Platform

Linux debian 5.10.0-16-amd64 #1 SMP Debian 5.10.127-2 (2022-07-23) x86_64 GNU/Linux

### What steps will reproduce the bug?

Run the following bunch of code

bun-fault.ts
```
console.log('Loading promises')
import { mkdtemp , rmdir } from 'fs/promises';

async function  testFS() {
    try {
        console.log("Running mkdtemp");
        const dname=await mkdtemp('foo-');
        console.log("tempdir: "+dname);
        await rmdir(dname);
    } catch (err) {
        console.error(err);
    }
}
testFS();

```
package.json:
```{
  "dependencies": {
    "@types/node": "^18.6.3"
  }
}
```
tsconfig.json:
```{
    // Most ts-node options can be specified here using their programmatic names.
    "ts-node": {
      "transpileOnly": false      
    },
  "compilerOptions": {
    "target": "es5",
    "esModuleInterop": true,
    "incremental": true
  }
}
```

### How often does it reproduce? Is there a required condition?

Every time

### What is the expected behavior?

Create a temp dir and remove it, testing await/async 

### What do you see instead?

```
$ bun bun-fault.ts 
Loading promises
Running mkdtemp

SegmentationFault at 0x0000000000000000


----- bun meta -----
Bun v0.1.10 (fe7180bc) Linux x64 #1 SMP Debian 5.10.127-2 (2022-07-23)
AutoCommand: 
Elapsed: 38ms | User: 24ms | Sys: 16ms
RSS: 67.11MB | Peak: 49.66MB | Commit: 67.11MB | Faults: 0
----- bun meta -----

Ask for #help in https://bun.sh/discord or go to https://bun.sh/issues
```

### Additional information

Running on Debian x86 machine (old Intel NUC Centrino).
I was unable to find a similar issue so far.

### PR DESCRIPTION
**Pull request by zhuzilin**
fix #1144

### CODE REVIEW

File: src/bun.js/node/node_fs.zig [hunk 1]
```diff
@@ -3013,12 +3013,15 @@ pub const NodeFS = struct {
...
         var prefix_buf = &this.sync_error_buf;
-        prefix_buf[0] = 0;
         const len = args.prefix.len;
         if (len > 0) {
             @memcpy(prefix_buf, args.prefix.ptr, len);
-            prefix_buf[len] = 0;
         }
+        var i: usize = 0;
+        while (i < 6) : (i += 1) {
+            prefix_buf[len + i] = 'X';
```
Comment (Jarred-Sumner) [inline src/bun.js/node/node_fs.zig]:
```suggestion
            prefix_buf[len..][0..6].* = "XXXXXX"
            prefix_buf[len..][6] = 0
```

Comment (Jarred-Sumner) [reply src/bun.js/node/node_fs.zig]: We can remove the loop and this code will be slightly faster most likely

**Comment by Jarred-Sumner** [comment]
LGTM after small tweak

File: src/bun.js/node/node_fs.zig (hunk 1 above)
Comment (zhuzilin) [reply src/bun.js/node/node_fs.zig]: fixed. didn't know zig has such syntax before :p

File: src/bun.js/node/node_fs.zig [hunk 2]
```diff
@@ -3013,12 +3013,12 @@ pub const NodeFS = struct {
 
     pub fn mkdtemp(this: *NodeFS, args: Arguments.MkdirTemp, comptime flavor: Flavor) Maybe(Return.Mkdtemp) {
         var prefix_buf = &this.sync_error_buf;
-        prefix_buf[0] = 0;
         const len = args.prefix.len;
         if (len > 0) {
             @memcpy(prefix_buf, args.prefix.ptr, len);
-            prefix_buf[len] = 0;
         }
+        prefix_buf[len..][0..6] = "XXXXXX";
```
Comment (Jarred-Sumner) [inline src/bun.js/node/node_fs.zig]:
```suggestion
        prefix_buf[len..][0..6] = "XXXXXX".*;
```

File: src/bun.js/node/node_fs.zig [hunk 3]
```diff
@@ -3013,12 +3013,12 @@ pub const NodeFS = struct {
 
     pub fn mkdtemp(this: *NodeFS, args: Arguments.MkdirTemp, comptime flavor: Flavor) Maybe(Return.Mkdtemp) {
         var prefix_buf = &this.sync_error_buf;
-        prefix_buf[0] = 0;
         const len = args.prefix.len;
```
Comment (Jarred-Sumner) [inline src/bun.js/node/node_fs.zig]: Edgecase:
```suggestion
        const len = @minimum(args.prefix.len, prefix_buf.len - 7);
```

**Pull request merged**